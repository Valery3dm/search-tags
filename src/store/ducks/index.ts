import { combineReducers } from 'redux';
import { itemReducer } from './itemReducer';

export const rootReducer = combineReducers({
    item: itemReducer
})

export type RootState = ReturnType<typeof rootReducer>
