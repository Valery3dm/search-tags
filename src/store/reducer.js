const initialState = {
  itemsList: [],
  listOfThreeLastItems: [],
  inputItem: ''
};

const FETCH_ITEMS = "FETCH_ITEMS";
const SET_THREE_LAST_ITEMS = "SET_THREE_LAST_ITEMS";
const SET_INPUT_ITEM = "SET_INPUT_ITEM";

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_ITEMS:
      return { ...state, itemsList: action.payload };
    case SET_THREE_LAST_ITEMS:
      return { ...state,
         listOfThreeLastItems: [ ...state.listOfThreeLastItems, action.payload]
        };
    case SET_INPUT_ITEM:
      return { ...state, inputItem: action.payload };
    default:
      return state;
  }
};

export default reducer;