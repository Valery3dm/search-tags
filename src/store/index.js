import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import createSagaMiddleware from "redux-saga";

import { duck } from "./ducks/widgets";


const sagaMiddleware = createSagaMiddleware();

const store = createStore(duck.reducers.reducer, composeWithDevTools(applyMiddleware(sagaMiddleware)));

export default store;

sagaMiddleware.run(duck.saga);
