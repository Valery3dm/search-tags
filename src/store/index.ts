import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import { sagaFetchItemsWatcher } from './saga/sagaFetchItems';
import { itemReducer } from './ducks/itemReducer';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  itemReducer,
  composeWithDevTools(applyMiddleware(sagaMiddleware))
);

export default store;

sagaMiddleware.run(sagaFetchItemsWatcher);