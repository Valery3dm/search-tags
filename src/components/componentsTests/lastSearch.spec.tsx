import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import store from '../../store';
import { BrowserRouter } from 'react-router-dom';

import { LastSearch } from '../index';



Enzyme.configure({ adapter: new Adapter() });

describe('Last-Search component', () => {
    const listOfThreeLastItems = ['item1','item2','item3'];

    it('LastSearch Snapshot component', () => {
        const component = mount(
            <Provider store={store}>
                <BrowserRouter>
                    <LastSearch listOfThreeLastItems={listOfThreeLastItems}/>
                </BrowserRouter>
            </Provider>
        );
        expect(toJson(component)).toMatchSnapshot();
    });
});
