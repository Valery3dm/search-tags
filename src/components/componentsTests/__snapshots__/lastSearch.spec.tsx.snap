// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Last-Search component LastSearch Snapshot component 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <LastSearch
        listOfThreeLastItems={
          Array [
            "item1",
            "item2",
            "item3",
          ]
        }
      >
        <styled.div>
          <div
            className="sc-jSFjdj bcarFw"
          >
            <div>
              <styled.ul>
                <ul
                  className="sc-gKAaRy fgCheT"
                >
                  <li
                    key="0"
                  >
                    <Link
                      to="/item1"
                    >
                      <LinkAnchor
                        href="/item1"
                        navigate={[Function]}
                      >
                        <a
                          href="/item1"
                          onClick={[Function]}
                        >
                          <Styled(Button)
                            onClick={[Function]}
                            variant="outline-dark"
                          >
                            <Button
                              active={false}
                              className="sc-eCApnc iAbNsN"
                              disabled={false}
                              onClick={[Function]}
                              variant="outline-dark"
                            >
                              <button
                                className="sc-eCApnc iAbNsN btn btn-outline-dark"
                                disabled={false}
                                onClick={[Function]}
                                type="button"
                              >
                                item1
                              </button>
                            </Button>
                          </Styled(Button)>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </li>
                  <li
                    key="1"
                  >
                    <Link
                      to="/item2"
                    >
                      <LinkAnchor
                        href="/item2"
                        navigate={[Function]}
                      >
                        <a
                          href="/item2"
                          onClick={[Function]}
                        >
                          <Styled(Button)
                            onClick={[Function]}
                            variant="outline-dark"
                          >
                            <Button
                              active={false}
                              className="sc-eCApnc iAbNsN"
                              disabled={false}
                              onClick={[Function]}
                              variant="outline-dark"
                            >
                              <button
                                className="sc-eCApnc iAbNsN btn btn-outline-dark"
                                disabled={false}
                                onClick={[Function]}
                                type="button"
                              >
                                item2
                              </button>
                            </Button>
                          </Styled(Button)>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </li>
                  <li
                    key="2"
                  >
                    <Link
                      to="/item3"
                    >
                      <LinkAnchor
                        href="/item3"
                        navigate={[Function]}
                      >
                        <a
                          href="/item3"
                          onClick={[Function]}
                        >
                          <Styled(Button)
                            onClick={[Function]}
                            variant="outline-dark"
                          >
                            <Button
                              active={false}
                              className="sc-eCApnc iAbNsN"
                              disabled={false}
                              onClick={[Function]}
                              variant="outline-dark"
                            >
                              <button
                                className="sc-eCApnc iAbNsN btn btn-outline-dark"
                                disabled={false}
                                onClick={[Function]}
                                type="button"
                              >
                                item3
                              </button>
                            </Button>
                          </Styled(Button)>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </li>
                </ul>
              </styled.ul>
            </div>
          </div>
        </styled.div>
      </LastSearch>
    </Router>
  </BrowserRouter>
</Provider>
`;
